File: src/main/java/org/tokend/sdk/api/v2/accounts/model/Policies.java
package org.tokend.sdk.api.v2.accounts.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import org.tokend.sdk.api.base.model.NameValue;

import java.util.List;

public class Policies {

    @JsonProperty("type_i")
    private Integer typeI;

    @JsonProperty("types")
    private List<? extends NameValue<Integer>> types;

    public Integer getTypeI() {
        return typeI;
    }

    public List<? extends NameValue<Integer>> getTypes() {
        return types;
    }
}

File: src/main/java/org/tokend/sdk/api/base/model/Policies.kt
package org.tokend.sdk.api.base.model

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Holds policies by bitmask
 */
data class Policies(
        /**
         * Policies parsed as [NameValue] pairs
         */
        @JsonProperty("flags")
        val flags: List<NameValue<Int>>?,

        /**
         * Policies bitmask
         */
        @JsonProperty("mask")
        val mask: Long
) {
    /**
     * @return true if given [policy] is set in bitmask,
     * false otherwise
     *
     * @see mask
     */
    fun have(policy: Long): Boolean {
        return (mask and policy) == policy
    }

    /**
     * @return true if given [policy] is set in bitmask,
     * false otherwise
     *
     * @see mask
     */
    fun have(policy: Int): Boolean {
        return have(policy.toLong())
    }
}